examples/fcma/mvpa_classification.py

```diff
@@ -0,0 +1,139 @@
+
+def prepare_data(raw_data, epoch_info):
+    """ prepare the data for model training and prediction
+
+    Average the activity within epochs and z-scoring within subject.
+
+    Parameters
+    ----------
+    raw\_data: list of 2D array
+        each element of the list is the raw data of a subject,
+        in the shape of [num_voxels, num_TRs]
+        Assuming all data have the same num_voxels
+
+    epoch\_info: list of tuple (label, sid, start, end).
+        label is the condition labels of the epochs;
+        sid is the subject id, corresponding to the index of raw_data;
+        start is the start TR of an epoch (inclusive);
+        end is the end TR of an epoch(exclusive).
+        Assuming len(labels) labels equals the number of epochs and
+        the epochs of the same sid are adjacent in epoch_info
+
+    Returns
+    -------
+    processed\_data: 2D array in shape [num_voxels, num_epochs]
+        averaged epoch by epoch processed data
+
+    labels: 1D array
+        contains labels of the data
+    """
+    num_epochs = len(epoch_info)
+    (d1, _) = raw_data[0].shape
+    processed_data = np.empty([d1, num_epochs])
+    labels = np.empty(num_epochs)
+    subject_count = [0]  # counting the epochs per subject for z-scoring
+    cur_sid = -1
+    # averaging
+    for idx, epoch in enumerate(epoch_info):
+        labels[idx] = epoch[0]
+        if cur_sid != epoch[1]:
+            subject_count.append(0)
+            cur_sid = epoch[1]
+        subject_count[-1] += 1
+        processed_data[:, idx] = \
+            np.mean(raw_data[cur_sid][:, epoch[2]:epoch[3]],
+                    axis=1)
+    # z-scoring
+    cur_epoch = 0
+    for i in subject_count:
+        if i > 1:
+            processed_data[:, cur_epoch:cur_epoch + i] = \
+                zscore(processed_data[:, cur_epoch:cur_epoch + i],
+                       axis=1, ddof=0)
+        cur_epoch += i
+    # if zscore fails (standard deviation is zero),
+    # set all values to be zero
+    processed_data = np.nan_to_num(processed_data)
+
+    return processed_data, labels
+
+# python mvpa_classification.py face_scene bet.nii.gz face_scene/visual_top_mask.nii.gz face_scene/fs_epoch_labels.npy 18
+if __name__ == '__main__':
+    data_dir = sys.argv[1]
+    extension = sys.argv[2]
+    mask_file = sys.argv[3]
+    epoch_file = sys.argv[4]
+
+    raw_data = []
+    mask_img = nib.load(mask_file)
+    mask = mask_img.get_data()
+    mask = mask.astype(np.bool)
+    logger.info(
+        'mask size: %d' %
+        np.sum(mask)
+    )
+    files = [f for f in sorted(os.listdir(data_dir))
+             if os.path.isfile(os.path.join(data_dir, f))
+             and f.endswith(extension)]
+    for f in files:
+        img = nib.load(os.path.join(data_dir, f))
+        data = img.get_data()
+        # apply mask
+        data = data[mask, :]
+        raw_data.append(data)
+        logger.info(
+            'file %s is loaded, with data shape %s' % (f, data.shape)
+        )
```

Comment:
Yes, I am also thinking of it, too. Currently, there is a slight difference between these two: one producing `nTRs*nVoxels` matrix in row-major; the other producing `nVoxels*nTRs` matrix in row-major. I will simplify the code (including moving `file_io.py` into brainiak) after some more thinking.